src/sprites/circle_sprite.cpp

```diff
@@ -0,0 +1,44 @@
+#include <sprites/circle_sprite.h>
+
+void CircleSprite::setSize(uint16_t x, uint16_t y) {
+    //Circle has equal sizes
+    Sprite::setSize(x, x);
+}
+
+uint16_t CircleSprite::getRadius() {
+    return Sprite::getSizeX(); 
+}
+
+void CircleSprite::setRadius(uint16_t r) {
+    setSize(r * 2, r * 2);
+}
+
+
+int16_t CircleSprite::getOffsetX() {
+    if (getAlignment() & SpriteAlignment::LEFT) {
+        return -(getSizeX() / 2);
+    } else if (getAlignment() & SpriteAlignment::RIGHT) {
+        return (getSizeX() / 2);
+    }
+
+    return 0;
+}
+
+int16_t CircleSprite::getOffsetY() {
+    if (getAlignment() & SpriteAlignment::TOP) {
+        return -(getSizeY() / 2);
+    } else if (getAlignment() & SpriteAlignment::BOTTOM) {
+        return getSizeY() / 2;
+    }
+
+    return 0;
+}
+
+void CircleSprite::draw(OswHal *hal) {
+    int16_t posX = getX() - getOffsetX(), posY = getY() - getOffsetY();
+    if (getFilled()) {
+        hal->getCanvas()->fillCircle(posX, posY, getSizeX() / 2, getColor());
```

Comment:
The recommended way is to use `hal->gfx()` instead of `hal->getCanvas()`

Reply:
Little confused on the purpose of gfx(). Canvas is already buffered and it looks like gfx is actually allocating a second frame buffer?

Reply:
Gfx(gfx_2d) is the buffer that canvas writes to. The desktop code only uses the drawfunctions of gfx_2d.

Reply:
A little confused why we the have gfx_2d. For the desktop emulation I think we should still be using `ArduinoGraphics2DCanvas` but for the `Arduino_G` passing it an adapter which writes to the framebuffer of a desktop application?

Reply:
sorry for the late reply...

yes, this way we can cross compile the code on desktop and do experiments using the same renderer using an SDL window (src: [matrix.cpp](https://github.com/Open-Smartwatch/lib-open-smartwatch/blob/master/examples/matrix/main.cpp)):

![demo](https://user-images.githubusercontent.com/930238/128916735-cd9d4ccc-d612-4e25-8b69-815389eb64c7.gif)